### PR TITLE
Add nrf54h20dk cpuppr configuration to SPI tests

### PR DESCRIPTION
**Pull request by nordic-krch**
Add configuration for `spi_loopback` and `spi_controller_peripheral` tests. Reworked configuration of `spi_error_cases` using same scheme of configuration as for other tests which does not require special case in `testcase.yaml`.

### CODE REVIEW

**Review by kl-cruz — DISMISSED** [review]

**Review by masz-nordic — DISMISSED** [review]

File: boards/nordic/nrf54h20dk/Kconfig.defconfig [hunk 1]
```diff
@@ -14,3 +14,11 @@ config BT_CTLR
...
 
 endif # BOARD_NRF54H20DK_NRF54H20_CPURAD
+
+if BOARD_NRF54H20DK_NRF54H20_CPUPPR
+
+# As PPR has limited memory most of tests does not fit with asserts enabled.
+config ASSERT
+	default n
+
+endif # BOARD_NRF54H20DK_NRF54H20_CPUPPR
```
Comment (gmarull) [inline boards/nordic/nrf54h20dk/Kconfig.defconfig:24]: why not enable this at soc level?, also, make it conditional to ztest maybe? like

```
config ASSERT
    default n if ZTEST
```

Comment (nordic-krch) [reply boards/nordic/nrf54h20dk/Kconfig.defconfig]: right, done for soc level.
As for assert kconfig. I used to have like you suggested but changed based on that comment: https://github.com/zephyrproject-rtos/zephyr/pull/74297#discussion_r1642541482

**Review by gmarull — DISMISSED** [review]

**Review by masz-nordic — DISMISSED** [review]

**Review by hakehuang — APPROVED** [review]

**Comment by nordic-krch** [comment]
@gmarull @masz-nordic @kl-cruz can you re-approve? I had to rebase.

**Review by gmarull — APPROVED** [review]

**Comment by nordic-krch** [comment]
@tbursztyka can you take a look?

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**